packages/labs/compiler/src/lib/template-transform.ts

```diff
@@ -242,6 +322,39 @@ class CompiledTemplatePass {
       },
     });
 
+    // Add required unique ctor identifiers for parts added.
+    const f = this.context.factory;
+    for (const part of parts) {
+      if (part.type === PartType.ATTRIBUTE) {
+        const ctorType = part.ctorType;
+        switch (ctorType) {
+          case PartType.ATTRIBUTE: {
+            this.attributePartConstructorNames.AttributePart ??=
+              f.createUniqueName('A');
+            break;
+          }
+          case PartType.BOOLEAN_ATTRIBUTE: {
+            this.attributePartConstructorNames.BooleanAttributePart ??=
+              f.createUniqueName('B');
+            break;
+          }
+          case PartType.EVENT: {
+            this.attributePartConstructorNames.EventPart ??=
+              f.createUniqueName('E');
+            break;
+          }
+          case PartType.PROPERTY: {
+            this.attributePartConstructorNames.PropertyPart ??=
+              f.createUniqueName('P');
+            break;
+          }
+          default: {
+            throw new Error(`Internal Error: Unexpected attribute type.`);
```

Comment:
Can add a `never` assignment here for the type system to help assert that we have all cases, once we have a specific type for the ctorType field

Reply:
Done! Nice!